substrate/frame/support/src/storage/mod.rs

```diff
@@ -1473,6 +1516,14 @@ impl<T: Encode> StorageDecodeLength for Vec<T> {}
 impl<T: Encode> StorageAppend<T> for BTreeSet<T> {}
 impl<T: Encode> StorageDecodeLength for BTreeSet<T> {}
```

Comment:
Yes we should not enable that behaviour.

Reply:
Should be removed

Reply:
Clippy didn't allow me to remove it because
```rust
// Blanket implementation StorageDecodeNonDedupLength
// for all types that are StorageDecodeLength.
impl<T: StorageDecodeLength> StorageDecodeNonDedupLength for T {
	fn decode_non_dedup_len(key: &[u8]) -> Option<usize> {
		T::decode_len(key)
	}
}
```

Reply:
The main objective of this MR is to remove the `StorageDecodeLength for BTreeSet` because it is wrong and confusing.  
Now as replacement we have `StorageDecodeNonDedupLength for BTreeSet`. I dont know what Clippy says, but its definitely possible.

Reply:
Yes ignore Clippy :)